analyzers/src/SonarAnalyzer.CSharp/Rules/TimestampsShouldNotBeUsedAsPrimaryKeys.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class TimestampsShouldNotBeUsedAsPrimaryKeys : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S3363";
+    private const string MessageFormat = "'{0}' should not be used as a type for primary keys";
+
+    private static readonly KnownType[] TemporalTypes = new[]
+    {
+        KnownType.System_DateOnly,
+        KnownType.System_DateTime,
+        KnownType.System_DateTimeOffset,
+        KnownType.System_TimeSpan,
+        KnownType.System_TimeOnly
+    };
+    private static readonly string[] KeyAttributeTypeNames = TypeNamesForAttribute(KnownType.System_ComponentModel_DataAnnotations_KeyAttribute);
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(start =>
+        {
+            if (!IsReferencingEntityFramework(start.Compilation))
+            {
+                return;
+            }
+
+            context.RegisterNodeAction(c =>
+            {
+                // To improve performance the attributes and property types are only matched by their names, rather than their actual symbol.
+                // This results in a couple of FNs, but those scenarios are very rare.
+                var classDeclaration = (ClassDeclarationSyntax)c.Node;
+                var className = classDeclaration.Identifier.ValueText;
+                var keyProperties = classDeclaration.Members
+                    .OfType<PropertyDeclarationSyntax>()
+                    .Where(x => IsPublicReadWriteProperty(x) && IsTemporalType(x) && IsKeyProperty(x, className));
```

Comment:
Fix identing
```suggestion
                    .Where(x => IsPublicReadWriteProperty(x) 
                                         && IsTemporalType(x) 
                                         && IsKeyProperty(x, className));
```

Reply:
It does not look good in the rendered suggestion.
It basically should go on separate lines and alined with the beginning of IsPublicReadWriteProperty.